l5kit/l5kit/sampling/agent_sampling.py

```diff
@@ -15,6 +15,94 @@
 from .slicing import get_future_slice, get_history_slice
 
 
+def get_agent_context(
+    state_index: int,
+    frames: np.ndarray,
+    agents: np.ndarray,
+    tl_faces: np.ndarray,
+    history_num_frames: int,
+    history_step_size: int,
+    future_num_frames: int,
+    future_step_size: int,
+) -> Tuple[np.ndarray, np.ndarray, List[np.ndarray], List[np.ndarray], List[np.ndarray], List[np.ndarray]]:
+    """
+    Slice zarr or numpy arrays to get the context around the agent onf interest (both in space and time)
+
+    Args:
+        state_index (int): frame index inside the scene
+        frames (np.ndarray): frames from the scene
+        agents (np.ndarray): agents from the scene
+        tl_faces (np.ndarray): tl_faces from the scene
+        history_num_frames (int): how many frames in the past to slice
+        history_step_size (int): size of each step in the past
+        future_num_frames (int): how many frames in the future to slice
+        future_step_size (int): size of each step in the future
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray, List[np.ndarray], List[np.ndarray], List[np.ndarray], List[np.ndarray]]
+    """
+
+    #  the history slice is ordered starting from the latest frame and goes backward in time., ex. slice(100, 91, -2)
+    history_slice = get_history_slice(state_index, history_num_frames, history_step_size, include_current_state=True)
+    future_slice = get_future_slice(state_index, future_num_frames, future_step_size)
+
+    history_frames = frames[history_slice].copy()  # copy() required if the object is a np.ndarray
+    future_frames = frames[future_slice].copy()
+
+    sorted_frames = np.concatenate((history_frames[::-1], future_frames))  # from past to future
+
+    # get agents (past and future)
+    agent_slice = get_agents_slice_from_frames(sorted_frames[0], sorted_frames[-1])
+    agents = agents[agent_slice].copy()
+    # sync interval with the agents array
+    history_frames["agent_index_interval"] -= agent_slice.start
+    future_frames["agent_index_interval"] -= agent_slice.start
+    history_agents = filter_agents_by_frames(history_frames, agents)
+    future_agents = filter_agents_by_frames(future_frames, agents)
+
+    # get traffic lights (past and future)
+    tl_slice = get_tl_faces_slice_from_frames(sorted_frames[0], sorted_frames[-1])
+    tl_faces = tl_faces[tl_slice].copy()
+    # sync interval with the traffic light faces array
+    history_frames["traffic_light_faces_index_interval"] -= tl_slice.start
+    history_tl_faces = filter_tl_faces_by_frames(history_frames, tl_faces)
+    future_tl_faces = filter_tl_faces_by_frames(future_frames, tl_faces)
+
+    return history_frames, future_frames, history_agents, future_agents, history_tl_faces, future_tl_faces
+
+
+def compute_agent_speed(
```

Comment:
```suggestion
def compute_agent_velocity(
```
We are computing velocity (v_x, v_y) and not speed (scalar)